src/components/ArticleList.jsx

```diff
@@ -1,28 +1,37 @@
 import { useEffect, useState } from "react";
 import { getArticles } from "../api";
 import ArticleCard from "./ArticleCard";
+import ReturnToMain from "./ReturnToMain";
 
 export default function ArticlesList() {
     const [articles, setArticles] = useState([]);
-  
+    const [loading, setIsLoading] = useState(true);
```

Comment:
we should start this as false, and set it to true at the start of the useEffect.